[ADD] Sales: loyalty and discount programs (#3587)
odoo/documentation: content/applications/sales/sales/products_prices/loyalty_discount.rst

=================================
Use discount and loyalty programs
=================================

The Odoo *Sales*, *eCommerce*, and *Point of Sale* applications allow users to create discount and
loyalty programs that customers can use for online and in-store shopping. These programs offer more
varied, public, and time-sensitive pricing options than :doc:`pricelists <prices/pricing>` do.

Configure the settings
======================

To begin using discount and loyalty programs, navigate to :menuselection:`Sales --> Configuration
--> Settings`. Under the :guilabel:`Pricing` heading, activate the :guilabel:`Discounts, Loyalty &
Gift Card` setting.

Configure discount and loyalty programs
=======================================

> To create discount and loyalty programs, go to :menuselection:`Sales --> Products --> Discount &
> Loyalty`. Then, choose one of the template cards, or click :guilabel:`New` to create a new program
> from scratch. Or, if there are already existing programs, select an existing program to edit it.

Reviewer: ```suggestion
To create discount and loyalty programs, go to :menuselection:`Sales app --> Products --> Discount &
Loyalty`. Then, choose one of the template cards, or click :guilabel:`New` to create a new program
from scratch. Or, if there are already existing programs, select an existing program to edit it.
```
just a quick reminder to mention that the templates will only be there if no other data is present...

.. image:: loyalty_discount/price-discount-loyalty.png
   :align: center
   :alt: Discount and loyalty program template cards.

.. note::

>    Templates disappear once the first program is created.

Reviewer: you could add the note re: Templates only being available on a completely blank (data-less) database here - just a thought

> The program form contains the following options:

Reviewer: ```suggestion
Creating (or editing an existing) program reveals a program form on a separate page. 

The program form contains the following options:
```

- :guilabel:`Program Name`: Enter the name of the program. The program name is *not* visible to the
  customer.
- :guilabel:`Program Type`: Select the desired :ref:`program type
  <pricing_management/program-types>`.
- :guilabel:`Currency`: Select the currency used for the program.

> - :guilabel:`Points Unit`: Enter the name of the points used for the :guilabel:`Loyalty Cards`
>   program (e.g. `Loyalty Points`). The points unit name is visible to the customer.

Reviewer: ```suggestion
- :guilabel:`Points Unit`: Enter the name of the points used for the :guilabel:`Loyalty Cards`
  program (e.g. `Loyalty Points`). The points unit name is visible to the customer. This field
  is only available when the :guilabel:`Program Type` is set to :guilabel:`Loyalty Cards`.
```
bit of elaboration explaining this field is only available when the Program type is set to "Loyalty Cards"

> - :guilabel:`Validity`: Select the date until which the program is valid. Leave this field blank for
>   no end date.

Reviewer: ```suggestion
- :guilabel:`Validity`: Select the date until which the program is valid. Leave this field blank for
  no end date. Thus, ensuring the program is always valid and doesn't expire.
```
just some minor elaboration on why some may choose to leave this field blank.

- :guilabel:`Limit Usage`: Check this box and enter a number to limit the number of times the
  program can be used during the :guilabel:`Validity` period.
- :guilabel:`Company`: In the case of multiple companies, choose the company for which the program
  is available.
- :guilabel:`Available On`: Select the app(s) on which the program is available.
- :guilabel:`Website`: Select the website(s) on which the program is available. Leave this field
  blank to make it available on all websites.
- :guilabel:`Point of Sale`: Select the point(s) of sale at which the program is available. Leave
  this field blank to make it available at all PoS.

.. image:: loyalty_discount/price-programs.png
   :align: center
   :alt: Program options

.. note::
   The options available on the program form vary depending on the :ref:`Program Type
   <pricing_management/program-types>` selected.

> All of the existing cards, codes, etc. that have been generated for the program are available
> through the smart button in the upper-right corner.

Reviewer: ```suggestion
All of the existing cards, codes, coupons, etc. that have been generated for the program are accessible,
via the corresponding smart button, located in the upper-right corner.
```

.. image:: loyalty_discount/price-programs-items.png
   :align: center
   :alt: Program items tab

.. _pricing_management/program-types:

Program types
-------------

> The different :guilabel:`Program Types` are:

Reviewer: ```suggestion
The different :guilabel:`Program Types` available on the program form are:
```

- :guilabel:`Coupons`: Generate and share single-use coupon codes that grant immediate access to
  rewards.

> - :guilabel:`Next Order Coupons`: Generate and share single-use coupon codes that grant access to
>   rewards on the next order.

Reviewer: ```suggestion
- :guilabel:`Next Order Coupons`: Generate and share single-use coupon codes that grant access to
  rewards on the customer's next order.
```

- :guilabel:`Loyalty Cards`: When making purchases, the customer accumulates points to exchange for
  rewards on future orders.
- :guilabel:`Promotions`: Set conditional rules for ordering products, which, when fulfilled, grant
  access to rewards for the customer.
- :guilabel:`Discount Code`: Set codes which, when entered upon checkout, grant discounts to the
  customer.
- :guilabel:`Buy X Get Y`: For every X item bought, the customer is granted 1 credit. After
  accumulating a specified amount of credits, the customer can trade them in to receive Y item.

Conditional rules
-----------------

Next, configure the :guilabel:`Conditional rules` that determine when the program applies to a
customer's order.

> In the :guilabel:`Rules & Rewards` tab, click :guilabel:`Add` next to :guilabel:`Conditional rules`
> to add *conditions*.

Reviewer: ```suggestion
In the :guilabel:`Rules & Rewards` tab, click :guilabel:`Add` next to :guilabel:`Conditional rules`
to add *conditions* to the program. Doing so reveals a :guilabel:`Create Conditional rules` pop-up window.
```

.. image:: loyalty_discount/price-conditional-rewards.png
   :align: center
   :alt: Conditional rewards

.. note::

>    The options for :guilabel:`Conditional rules` vary depending on the selected :ref:`Program Type
>    <pricing_management/program-types>`.

Reviewer: ```suggestion
   The following options for :guilabel:`Conditional rules` vary depending on the selected :ref:`Program Type
   <pricing_management/program-types>`.
```

> The following options are available for configuring conditional rules:

Reviewer: ```suggestion
The following options are available in the pop-up window for configuring conditional rules:
```

> - :guilabel:`Discount Code`: Enter a custom code used for the :guilabel:`Discount Code` program, or
>   use the default one generated by Odoo.

Reviewer: ```suggestion
- :guilabel:`Discount Code`: Enter a custom code to be used for the :guilabel:`Discount Code` program, or
  use the default one generated by Odoo.
```

- :guilabel:`Minimum Quantity`: Enter the minimum number of products that must be purchased in order
  to access the reward. Set the minimum quantity to at least `1` to ensure that the customer must
  make a purchase in order to access the reward.
- :guilabel:`Minimum Purchase`: Enter the minimum amount (in currency), with :guilabel:`tax
  Included` or :guilabel:`tax Excluded`, that must be spent in order to access the reward. If both a
  minimum quantity *and* minimum purchase amount are entered, then the customer's order must meet
  both conditions.
- :guilabel:`Products`: Select the product(s) for which the program applies.
- :guilabel:`Categories`: Select the category of products for which the program applies. Choose

>   :guilabel:`All` to apply it to all categories.

Reviewer: ```suggestion
  :guilabel:`All` to apply it to all product categories.
```

> - :guilabel:`Product Tag:` Select a tag to apply the program to products with that tag.

Reviewer: ```suggestion
- :guilabel:`Product Tag:` Select a tag to apply the program to products with that specific tag.
```

- :guilabel:`Grant`: Enter the number of points the customer earns :guilabel:`per order`,
  :guilabel:`per currency spent`, or :guilabel:`per unit paid` (for the :guilabel:`Loyalty Cards`
  and :guilabel:`Buy X Get Y` programs).

> 

Reviewer: Perhaps it would be useful to mention the "Save & Close" and "Save & New" button options available on the pop-up...What do you think?

.. image:: loyalty_discount/price-conditions.png
   :align: center
   :alt: Conditional rules to get rewards through a program type

Rewards
-------

In the :guilabel:`Rules & Rewards` tab of the program form, click :guilabel:`Add` next to

> :guilabel:`Rewards` to add *rewards*.

Reviewer: ```suggestion
:guilabel:`Rewards` to add *rewards* to the program. Doing so reveals a :guilabel:`Create Rewards` pop-up window.
```

.. note::

>    The options for :guilabel:`Rewards` vary depending on the selected :ref:`Program Type
>    <pricing_management/program-types>`.

Reviewer: ```suggestion
   The following options for :guilabel:`Rewards` vary depending on the selected :ref:`Program Type
   <pricing_management/program-types>`.
```

The following options are available for configuring rewards:

- :guilabel:`Reward Type`: Select the reward type among :guilabel:`Free Product`,
  :guilabel:`Discount`, and :guilabel:`Free Shipping`. The other options for reward configuration
  depend on the :guilabel:`Reward Type` selected.

  - :guilabel:`Free Product` Reward Type:

    - :guilabel:`Quantity Rewarded`: Select the number of free products rewarded to the customer.
    - :guilabel:`Product`: Select the product given for free as a reward. Only one product can be
      selected.
    - :guilabel:`Product Tag`: Select a tag to further specify the free product eligible for the
      reward.

  - :guilabel:`Discount` Reward Type:

    - :guilabel:`Discount`: Enter the discounted amount in either :guilabel:`percentage`,
      :guilabel:`currency per point`, or :guilabel:`currency per order`. Then, select whether the
      discount applies to the entire :guilabel:`Order`, only the :guilabel:`Cheapest Product` on the

>       order, or only :guilabel:`Specific Products`.

Reviewer: ```suggestion
      order, or only :guilabel:`Specific Products` on the order.
```

- :guilabel:`Points Unit`: Enter the name of the points used for the :guilabel:`Loyalty Cards`
  program (e.g. `Loyalty Points`). The points unit name *is* visible to the customer. This field is
  only available when the :guilabel:`Program Type` is set to :guilabel:`Loyalty Cards`.
- :guilabel:`Validity`: Select the date until which the program is valid. Leave this field blank for
  no end date, meaning the program is always valid and does not expire.
- :guilabel:`Limit Usage`: Check this box and enter a number to limit the number of times the
  program can be used during the :guilabel:`Validity` period.
- :guilabel:`Company`: In the case of multiple companies, choose the company for which the program
  is available.
- :guilabel:`Available On`: Select the app(s) on which the program is available.
- :guilabel:`Website`: Select the website(s) on which the program is available. Leave this field
  blank to make it available on all websites.
- :guilabel:`Point of Sale`: Select the point(s) of sale at which the program is available. Leave
  this field blank to make it available at all :abbr:`PoS (Point of Sale)`.

.. image:: loyalty_discount/price-programs.png
   :align: center

>    :alt: Program options

Reviewer 2: Please add more contextual information to the alt tags throughout the doc as well as a period
```suggestion
   :alt: Program options on the program form in Odoo.
```

window.

.. image:: loyalty_discount/price-conditional-rewards.png
   :align: center
   :alt: Conditional rewards

.. note::
   The options for :guilabel:`Conditional rules` vary depending on the selected :ref:`Program Type
   <pricing_management/program-types>`.

The following options are available for configuring conditional rules:

- :guilabel:`Discount Code`: Enter a custom code to be used for the :guilabel:`Discount Code`
  program, or use the default one generated by Odoo. This field is only available when the
  :guilabel:`Program Type` is set to :guilabel:`Discount Code`.
- :guilabel:`Minimum Quantity`: Enter the minimum number of products that must be purchased in order
  to access the reward. Set the minimum quantity to at least `1` to ensure that the customer must
  make a purchase in order to access the reward.
- :guilabel:`Minimum Purchase`: Enter the minimum amount (in currency), with :guilabel:`tax
  Included` or :guilabel:`tax Excluded`, that must be spent in order to access the reward. If both a
  minimum quantity *and* minimum purchase amount are entered, then the customer's order must meet

>   both conditions.

Reviewer 3: @samueljlieber where?
It's not obvious to me.

>   to access the reward. Set the minimum quantity to at least `1` to ensure that the customer must
>   make a purchase in order to access the reward.
> - :guilabel:`Minimum Purchase`: Enter the minimum amount (in currency), with :guilabel:`tax
>   Included` or :guilabel:`tax Excluded`, that must be spent in order to access the reward. If both a
>   minimum quantity *and* minimum purchase amount are entered, then the customer's order must meet
>   both conditions.

Reviewer 2: Sorry @StraubCreative & @meng-odoo I should have been clearer - On the **Minimum Qantity** *and* **Minimum Purchase** amount. 
```suggestion
  to access the reward. Set the minimum quantity to at least `1` to ensure that the customer must
  make a purchase in order to access the reward.
- :guilabel:`Minimum Purchase`: Enter the minimum amount (in currency), with :guilabel:`tax
  Included` or :guilabel:`tax Excluded`, that must be spent in order to access the reward. If both a
  :guilabel:`Minimum Quantity` *and* :guilabel:`Minimum Purchase` amount are entered, then the customer's order must meet
  both conditions.
```

>   both conditions.

Reviewer 3: eh, this is debatable since we're referring to the field values (input) not the labels, necessarily.
Could go either way imo 🤷

>   both conditions.

Reviewer 4: This was my thought as well re: Zac.

- :guilabel:`Grant`: Enter the number of points the customer earns :guilabel:`per order`,
  :guilabel:`per currency spent`, or :guilabel:`per unit paid` (for the :guilabel:`Loyalty Cards`
  and :guilabel:`Buy X Get Y` programs).

.. image:: loyalty_discount/price-conditions.png
   :align: center
   :alt: Conditional rules to get rewards through a program type

Click :guilabel:`Save & Close` to save the rule and close the pop-up window, or click
:guilabel:`Save & New` to save the rule and immediately create a new one.

Rewards
-------

In the :guilabel:`Rules & Rewards` tab of the program form, click :guilabel:`Add` next to
:guilabel:`Rewards` to add *rewards* to the program. This reveals a :guilabel:`Create Rewards`
pop-up window.

.. note::
   The options for :guilabel:`Rewards` vary depending on the selected :ref:`Program Type
   <pricing_management/program-types>`.

The following options are available for configuring rewards:

- :guilabel:`Reward Type`: Select the reward type among :guilabel:`Free Product`,
  :guilabel:`Discount`, and :guilabel:`Free Shipping`. The other options for reward configuration
  depend on the :guilabel:`Reward Type` selected.

>   - :guilabel:`Free Product` Reward Type:
> 

Reviewer 2: I think reiterating for each bullet of the reward type's as "X Reward Type" is redundant, when it is already implied by the parent bullet. 

I suggest formatting the reward type bullets as just "X" 

```suggestion
  - :guilabel:`Free Product`:

```

=============================
Discount and loyalty programs
=============================

The Odoo *Sales*, *eCommerce*, and *Point of Sale* applications allow users to create discount and
loyalty programs that customers can use for online and in-store shopping. These programs offer more
varied, public, and time-sensitive pricing options than :doc:`pricelists
</applications/sales/sales/products_prices/prices/pricing>`.

Configure the settings
======================

To begin using discount and loyalty programs, navigate to :menuselection:`Sales --> Configuration
--> Settings`. Under the :guilabel:`Pricing` heading, activate the :guilabel:`Discounts, Loyalty &

> Gift Card` setting by checking the box next to the feature. Finally, click :guilabel:`Save` to save

Reviewer 3: specificity
```suggestion
Gift Card` setting by checking the box next to the feature's label. Finally, click :guilabel:`Save` to save
```

the changes.

Configure discount and loyalty programs
=======================================

To create discount and loyalty programs, go to :menuselection:`Sales --> Products --> Discount &
Loyalty`.

If no discount or loyalty programs have been created yet, Odoo provides a choice of templates to
help create the first program. Choose one of the template cards, or click :guilabel:`New` to create
a new program from scratch.

Or, if there are already existing programs, select an existing program to edit it.

.. image:: loyalty_discount/price-discount-loyalty.png
   :align: center
   :alt: Discount and loyalty program template cards.

.. note::
   Templates only appear when no programs have been created, and they disappear once the first
   program is created.

Creating or editing a program opens the program form.

The program form contains the following options:

> - :guilabel:`Program Name`: Enter the name of the program. The program name is *not* visible to the

Reviewer 3: This is a matter of style, and depends on which guide you follow (i.e. APA vs. Chicago, etc.), however we've been leaving the first word after a colon *uncapitalized* and do not add periods at the end of list items.

However, what matters most is consistency, so the way this list is formatted works, as well 😉  
No need to change if you don't want to, but thought it was worth mentioning.
(The form walk-through is nice btw!)

  customer.
- :guilabel:`Program Type`: Select the desired :ref:`program type
  <sales/pricing_management/program-types>`.
- :guilabel:`Currency`: Select the currency used for the program.
- :guilabel:`Points Unit`: Enter the name of the points used for the :guilabel:`Loyalty Cards`

>   program (e.g. `Loyalty Points`). The points unit name *is* visible to the customer. This field is

Reviewer 3: suggestion to use a conjunction here (optional)
```suggestion
  program (e.g. `Loyalty Points`). The points unit name *is* visible to the customer and this field is
```

  no end date, meaning the program is always valid and does not expire.
- :guilabel:`Limit Usage`: Check this box and enter a number to limit the number of times the
  program can be used during the :guilabel:`Validity` period.
- :guilabel:`Company`: In the case of multiple companies, choose the company for which the program
  is available.
- :guilabel:`Available On`: Select the app(s) on which the program is available.
- :guilabel:`Website`: Select the website(s) on which the program is available. Leave this field
  blank to make it available on all websites.
- :guilabel:`Point of Sale`: Select the point(s) of sale at which the program is available. Leave
  this field blank to make it available at all :abbr:`PoS (Point of Sale)`.

.. image:: loyalty_discount/price-programs.png
   :align: center
   :alt: Program options on the loyalty program form.

.. note::
   The options available on the program form vary depending on the :ref:`Program Type
   <sales/pricing_management/program-types>` selected.

All of the existing cards, codes, coupons, etc. that have been generated for the program are

> accessible through the smart button located in the upper-right corner of the form.

Reviewer 3: specificity
```suggestion
accessible through the :guilabel:`Promos` smart button located in the upper-right corner of the form.
```

Reviewer 4: The name of this smart button actually changes depending on the program type, so we don't want to be specific here.
